identity.go

```diff
@@ -57,16 +57,22 @@ func (i Identity) Name() string {
 }
 
 // Whois returns a whois description for non-admin users.
-// TODO: Add optional room context?
-func (i Identity) Whois() string {
+func (i Identity) Whois(room *chat.Room) string {
 	fingerprint := "(no public key)"
 	if i.PublicKey() != nil {
 		fingerprint = sshd.Fingerprint(i.PublicKey())
 	}
+	awayMsg := ""
+	if m, ok := room.MemberByID(i.ID()); ok {
+		if m.IsAway() {
+			awayMsg = message.Newline + " > away since: " + humantime.Since(m.LastActivity())
```

Comment:
right now `away` is turned off when the user sends a message to the room, so i setup away to record when the user sent the last message instead of when the user set themselves as `away`. If the user joined the server and didn't send a message then their `away` since metric would be since they joined.

I tried to mask the sensitive information by making `away since` the publicly visible information, does that work?

since that's not clear from the code, do you want me to change the name of the APIs or add comments?

away message is a good idea, let me add it.

Reply:
The point of `/away` is to explicitly set state, not to leak implicit state. We don't want it to turn off automatically, and we don't want it to leak whether someone has been DM'ing with a friend quiety this whole time if they want to appear as away.

If I want to appear away, I should be able to appear away regardless of whether I'm actually there, that's how chat statuses have always worked and I'd prefer not to break that expectation.

Which part is not clear from the code? Comments are always welcome. :)

Reply:
changing: away needs to be updated explicitly.

was not clear: my intent of automatically updating away state. next commit will need users to explicitly manage away

Reply:
changing: away needs to be updated explicitly.

was not clear: my intent of automatically updating away state. next commit will need users to explicitly manage away